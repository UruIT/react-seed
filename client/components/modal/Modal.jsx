import React from 'react';
import style from './Modal.scss';
import PropTypes from 'prop-types';

const close = (e, onClose) => {
	e.preventDefault();
	if (onClose) {
		onClose();
	}
};

const Modal = ({ isOpen, onClose, children, closeClickingOutside }) => {
	if (!isOpen) {
		return null;
	}
	return (
		<div>
			<div className={style.modal}>
				{children}
			</div>
			<div className={style.backdrop} onClick={e => (closeClickingOutside ? close(e, onClose) : {})} />
		</div>
	);
};

Modal.propTypes = {
	isOpen: PropTypes.bool.isRequired,
	onClose: PropTypes.func.isRequired,
	children: PropTypes.node,
	closeClickingOutside: PropTypes.bool.isRequired
};

export default Modal;
