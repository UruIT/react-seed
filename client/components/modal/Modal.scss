.modal {
	position: fixed;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	z-index: 9999;
	background: #e9e8f1;
	border: solid 1px #0a0817;
	border-radius: 4px;
	padding: 24px;

	// size
	height: 100%;
	width: 100%;
	overflow-y: auto;
	overflow-x: auto;
}

.backdrop {
	position: fixed;
	top: 0px;
	left: 0px;
	width: 100%;
	height: 100%;
	z-index: 9998;
	background: rgba(0, 0, 0, 0.3);
}

@media (min-width: 448px) {
	.modal {
		height: auto;
		max-height: calc(100vh - 200px);
		width: auto;
		max-width: calc(100vw - 200px);
	}
}
