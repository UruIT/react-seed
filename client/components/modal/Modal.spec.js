import React from 'react';
import Modal from './Modal';
import renderer from 'react-test-renderer';

test('renders Modal', () => {
	const tree = renderer.create(<Modal />).toJSON();

	expect(tree).toMatchSnapshot();
});
