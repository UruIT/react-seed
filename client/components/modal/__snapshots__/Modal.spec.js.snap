// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal 1`] = `null`;
